core/charm/repository/charmhub.go

```diff
@@ -294,6 +334,45 @@ func (c *CharmHubRepository) retryResolveWithPreferredChannel(charmURL *charm.UR
 	}, nil
 }
 
+func transformRefreshResult(curl *charm.URL, refreshResult transport.RefreshResponse) (corecharm.EssentialMetadata, error) {
+	if refreshResult.Entity.MetadataYAML == "" {
+		return corecharm.EssentialMetadata{}, errors.Errorf("charmhub refresh response for %q does not include the contents of metadata.yaml", curl)
```

Comment:
This feels like it should be a not valid error.

Reply:
agreed.